### PR TITLE
CRS::Fuse: derive mount path from Record.Room, if available

### PR DESCRIPTION
**Pull request by Kunsi**
Please double-verify if this actually does the correct thing. I'm not 100% sure if that'll work as expected every time, because i'm not a perl expert.

### CODE REVIEW

**Comment by a-tze** [comment]
Hi, looks completely right on a first glance. I will check if perl has gotten some kind of coalesce operator in the last decade (i kind of remember something) that could dissolve the if-cascading.

As there is no doc for Record.Room: where does this propery come from, if it is present? Is it to resolve problems with "ugly" Fahrplan.Room names?

**Comment by Kunsi** [comment]
`Record.Room` is set by the mount4cut script, it contains the "normalized" room name (so we don't have funny behaviour when using room names containing slashes and such). See https://github.com/crs-tools/crs-scripts/blob/master/scripts/script-B-mount4cut.pl#L131

**Comment by Kunsi** [comment]
Will do some testing next week.

File: lib/CRS/Fuse.pm [hunk 1]
```diff
@@ -109,12 +109,19 @@ sub getFuseMounts {
...
-	my ($self, $vid) = @_;
+	my ($self, $vid, $room) = @_;
 	return unless defined($vid);
 	my $base = $self->{'paths'}->getPath('Raw');
 	die "ERROR: Processing.Path.Raw is not defined!\n" unless defined $base;
-	if (defined($self->{'Project.Slug'}) && defined($self->{'Fahrplan.Room'})) {
-		return $base . '/' . $self->{'Project.Slug'} . '/' . $self->{'Fahrplan.Room'} . "/$vid";
+	if (!defined($room)) {
+		if (defined($self->{'Record.Room'})) {
+			my $room = $self->{'Record.Room'};
```
Comment (fkusei) [inline lib/CRS/Fuse.pm]: This must not be `my ...`, otherwise it works fine.

**Comment by fkusei** [comment]
Note for myself, next week: Needs adjusting of the encoding profile, exchange the hardcoded `Fahrplan.Room` in the `raw` area with
```
    <xsl:choose>
      <xsl:when test="string-length($properties/property[@name='Record.Room']) &gt; 0">
        <xsl:value-of select="$properties/property[@name='Record.Room']"/>
      </xsl:when>
      <xsl:otherwise>
        <xsl:value-of select="$properties/property[@name='Fahrplan.Room']"/>
      </xsl:otherwise>
    </xsl:choose>
```

**Comment by fkusei** [comment]
needs `Record.Room` stuff for repair, too.

**Comment by Kunsi** [comment]
rebased to current master to resolve merge conflicts in `script-B-mount4cut`

**Comment by Kunsi** [comment]
> Note for myself, next week: Needs adjusting of the encoding profile, exchange the hardcoded `Fahrplan.Room` in the `raw` area with
> 
> ```
>     <xsl:choose>
>       <xsl:when test="string-length($properties/property[@name='Record.Room']) &gt; 0">
>         <xsl:value-of select="$properties/property[@name='Record.Room']"/>
>       </xsl:when>
>       <xsl:otherwise>
>         <xsl:value-of select="$properties/property[@name='Fahrplan.Room']"/>
>       </xsl:otherwise>
>     </xsl:choose>
> ```

For VOC, this means HD-Master profiles older than r21 are not compatible.

**Comment by a-tze** [comment]
Edited and merged as 6f86029

**Comment by a-tze** [comment]
Sorry, cannot mark it as merged, github is ignoring that feature request for at least 10 years now...

**Pull request closed without merging**